base-template.sh

```diff
@@ -250,10 +253,23 @@ execute_global_shared_hooks() {
 # Returns:
 #   1 if the hooks fail, 0 otherwise
 #####################################################
-execute_local_shared_hooks() {
+execute_all_shared_hooks() {
+    # track all executed hooks, to reject double execution
+    EXECUTED_SHARED_HOOKS=""
+
     if [ -f "$(pwd)/.githooks/.shared" ]; then
-        SHARED_HOOKS=$(grep -E "^[^#].+$" <"$(pwd)/.githooks/.shared")
-        process_shared_hooks "$SHARED_HOOKS" "$@" || return 1
+        SHARED_HOOKS=$(grep -E "^ *[^#].+$" <"$(pwd)/.githooks/.shared")
```

Comment:
Hm, why this change? Is it to filter out comment lines that (accidentally?) have spaces in front of the `#` ?

Reply:
Jup. Just to make it more agnostic?